### ISSUE TITLE
Implement SiteImprove Tokens and API Key

### ISSUE THREAD

**Issue by EWashb** [department-of-veterans-affairs/va.gov-cms#10275]
## Background

SiteImprove logins, Token, and API Key will be provided by the VHA Digital Media team/Blake Scates for the benefits hub product. Once we receive these critical pieces, we will need to login and implement for [CMS Deeplinking](https://help.siteimprove.com/support/solutions/articles/80000448272-what-is-cms-deeplinking-). 

## Acceptance Criteria

- [x] Coordinate login, API Key, and Tokens with Blake Scates (VHA Digital Media)

- [Enable deep linking ](https://help.siteimprove.com/support/solutions/articles/80000448431).  Request with two (2) examples, each example with: (1) Live URL and (2) CMS URL
If it is determined that deeplinking integration will be more involved outside of receiving the login, token, and API key, a new ticket should be created just for the integration.

**Comment by EWashb** [comment]
This is blocked August 15-19 due to Blake Scates being OOO

**Comment by EWashb** [comment]
@swirtSJW I'm going to add you here just to be the one to do the deep linking enablement. However, this is blocked until I can get you the proper login, token, and API key

**Comment by EWashb** [comment]
My CAG is down and blocking this issue. 

Dawn Emailed Blake Scates on 8/22/22 for update: 

Good afternoon Blake!

 

Erika is experiencing issues logging into CAG so I wanted to reach out to follow up from our last SiteImprove meeting.

 

Will we be able to receive the API Key, tokens, and/or login credentials? I know you needed to check with Jenny on these items first. Please let me know if there’s anything else our team can do to assist.

 

Thank you Blake!

 

Very Respectfully,

 

Dawn Pruitt

Delivery Manager, Sitewide CMS

Agile Six

**Comment by EWashb** [comment]
Dawn has also reached out in teams and followed up via email, but no response as of 8/25 at 9am

**Comment by EWashb** [comment]
Blake responded on 8/25:
<img width="733" alt="Screen Shot 2022-08-29 at 4 20 51 PM" src="https://user-images.githubusercontent.com/90111970/187301159-6799c51d-091c-404b-abc5-c6ee1301ab64.png">


Followed up again today 8/29 to see if JHL has made a decision

**Comment by EWashb** [comment]
Login credentials have been given to @dawnpruitt. However, we suppose that the best person to steward this login information and deep linking process is @swirtSJW. Dawn, please work with Steve to get him the login information.

Blake Scates told me on 9/6 that he will generate the API key and email. I will update this ticket and work with Steve once I receive that email from Blake

**Comment by EWashb** [comment]
We now have the API key from Blake Scates.

**Comment by dawnpruitt** [comment]
Received API key too late to implement in Sprint 66, moving to Sprint 67.

**Comment by EWashb** [comment]
@dawnpruitt we might want to get an update on this from Blake soon

**Comment by EWashb** [comment]
Update from Blake Scates today: SI costs have been approved by VHA Digital Media. CMS team awaiting next steps to get tokens/API key

**Comment by EWashb** [comment]
There is another blocker for this effort. There are concerns that implementing SiteImprove within the CMS would require FedRAMP approval. The current use of SiteImprove by VHA Digital Media does not have the same level of security concerns that implementing the API key in Drupal does. We will have a meeting with VHA Digital media, Dave C., and the Sitewide CMS team on 1/19/23

**Comment by EWashb** [comment]
We have a possible path forward with VHA DM looking to get a free trial token for testing purposes. If that test is successful, we will work to have a permanent token integration as early as March 2024

**Comment by maortiz-27-80** [comment]
@ariperez - can you attempt again with the following API username and API key:

User Name: Laura Flannery

Key: 5963c49ba007c022f19f029f3b789397

cc: @EWashb @BerniXiongA6 @ndouglas @laflannery @JunTaoLuo

**Comment by maortiz-27-80** [comment]
@ariperez - note 12/18 - Product is working with VHA-DM and Siteimprove re: Laura's role/permissions that _should_ unblock us to get the API keys working.  We'll provide updates in this ticket. cc: @BerniXiongA6 @EWashb @JunTaoLuo

**Comment by maortiz-27-80** [comment]
@ariperez - We have a brand new username and key provided.  See below:

Username: blake.scates@va.gov

API Key: 5963c49ba007c022f19f029f3b789397

This may also be already complete by you, but noted by Siteimprove: You must have the regular Plugin installed first before installing Prepublish:

1. [Drupal Plugin](https://www.drupal.org/docs/contributed-modules/siteimprove/installation-and-basic-configuration)
2. [Drupal Prepublish](https://www.drupal.org/node/3180424)

cc: @BerniXiongA6 @EWashb  @JunTaoLuo @laflannery @ndouglas

**Comment by maortiz-27-80** [comment]
Re: deeplinking.  Per Siteimprove 12/19:

We would need you to provide 2 sets of CMS editing URLs and their live page counterparts. You can use any two live pages - but not your homepage.

As soon as I receive that information back I will be able to proceed setting this up for you. 

1. [What is CMS deep linking?](https://help.siteimprove.com/support/solutions/articles/80000448272-what-is-cms-deeplinking-)
2. [How do I enable deep linking?](https://help.siteimprove.com/support/solutions/articles/80000448431-how-do-i-enable-cms-deeplinking-) 

cc: @ariperez @EWashb @BerniXiongA6

**Comment by ariperez** [comment]
@maortiz-27-80 , after implementing and verifying that the SiteImprove module does load, we discovered that the decoupled nature of our Drupal architecture does not allow prepublish checks to successfully run.

Initially, we assumed that the plugin would leverage data from Drupal's Edit page for these checks. However, we discovered that it actually depends on URLs generated for a page's Preview server. This reliance is less than ideal for our workflow, as it necessitates a content build prior to running any prepublish checks.

Additionally, and perhaps more critically, the module lacks a feature to specify the URL for SiteImprove to conduct the prepublish checks so we couldn't even point it to the content-build version. This oversight limits the module's utility in our current setup.

fyi @ndouglas

**Comment by maortiz-27-80** [comment]
@ariperez - Thank you for the summary!  @EWashb - Let's chat when you're back in the office for next steps.  thanks! - Marcia cc: @BerniXiongA6 @ndouglas @edmund-dunn @JunTaoLuo

**Comment by BerniXiongA6** [comment]
Closing this ticket since the tokens and API key implementation is complete. We've created a new ticket for deep linking (#16521 ). cc: @ariperez @maortiz-27-80

**Comment by maortiz-27-80** [comment]
Note: the root cause/incompatibility of our decoupled Drupal setup with the Siteimprove module could potentially be resolved in the future with next build replacing content build per nate (but that is a heavy lift for 2024 and beyond) - this is a dependency on the Accelerated Publishing team

**Comment by EWashb** [comment]
@ariperez @maortiz-27-80 did you all consider the frontend domain plugin? Do you have an understanding of why it didn't meet our needs? I want to make sure I'm giving VHA DM the right information on how to proceed. If we truly can do something with AP/CMS in the future or if we will continue having issues with Siteimprove's API regardless. If it is a Drupal module issue perhaps that is something we can control later on.
